src/components/trade/order/NewOrder.svelte

```diff
@@ -127,7 +127,21 @@
 	}
 
 	$: checkWarning($prices[$selectedMarket], $price, $orderType, $isLong);
-
+	
+  	// reset inputs on market change
+	function resetOrderFields() {
+    		setPrice();
+    		price.set();
+    		size.set();
+    		tpPrice.set();
+    		slPrice.set();
+    		hasTPSL.set(false);
+		isReduceOnly.set(false)
+    		isProtectedOrder.set(false);
+	}
+	
+  	selectedMarket.subscribe(resetOrderFields);
```

Comment:
Nothing wrong with this notation. However to be consistent with how the rest of subscriptions are done in the app, we can probably do

`$: resetOrderFields($selectedMarket)`